projects/controllers/logs/logs.py

```diff
@@ -251,49 +249,54 @@ def log_stream(self, pod, container):
-        if deployment_id is not None:
-            # Tries to retrieve any pods associated to a seldondeployment
-            pods.extend(
-                list_deployment_pods(deployment_id=deployment_id),
-            )
-
-        if len(pods) == 0:
-            # BUG: workflows and pods are deleted after 1 day due to a
-            # "Garbage Collector" feature of Argo workflows/Kubeflow Pipelines.
-            # The link below provide useful information on how to configure log
-            # persistence and where to set the Time-to-live (TTL) for workflows:
-            # https://github.com/kubeflow/pipelines/issues/844#issuecomment-559841627
-            # We can make use of these configurations and fix this bug later.
-            pods.extend(
-                list_workflow_pods(run_id=run_id),
-            )
-
-        iterators = list()
+        """
+        Search for online pods to start log stream
 
-        if not pods:
-            raise BadRequest("Unable to create log stream. No active pod available.")
+        Parameters
+        ----------
+            deployment_id: str
+            experiment_id: str
 
-        for pod in pods:
-            for container in pod.spec.containers:
+        Return
+        ------
+            Iterator
+        """
+        while True:
+            pods = list()
+            run_id = get_latest_run_id(experiment_id or deployment_id)
+            if deployment_id is not None:
+                # Tries to retrieve any pods associated to a seldondeployment
+                pods.extend(
+                    list_deployment_pods(deployment_id=deployment_id),
+                )
 
-                if container.name not in EXCLUDE_CONTAINERS:
-                    iterators.append(self.log_stream(pod, container))
+            if len(pods) == 0:
+                # BUG: workflows and pods are deleted after 1 day due to a
+                # "Garbage Collector" feature of Argo workflows/Kubeflow Pipelines.
+                # The link below provide useful information on how to configure log
+                # persistence and where to set the Time-to-live (TTL) for workflows:
+                # https://github.com/kubeflow/pipelines/issues/844#issuecomment-559841627
+                # We can make use of these configurations and fix this bug later.
+                pods.extend(
+                    list_workflow_pods(run_id=run_id),
+                )
 
-        return merge(iterators)
+            if not pods:
+                """
+                If the pods list is empty, it tries again to retrieve a pod list
+                """
+                pass
+            else:
+                iterators = list()
+                for pod in pods:
+                    for container in pod.spec.containers:
+                        if container.name not in EXCLUDE_CONTAINERS:
+                            iterators.append(self.log_stream(pod, container))
+                return merge(iterators)
```

Comment:
@dfvneto (acho que já conversamos sobre isso antes) para experimentos essa solução não funciona bem :(
Os pods são criados conforme o fluxo avança, e uma vez que e entra no loop `for pod in pods` novos pods não irão aparecer nos logs (a menos que se reabra a conexão.

**Tudo bem deixar assim, por enquanto.** Na próxima sprint, melhoramos a API para que a lista de iterators seja dinâmica, conforme os pods/containers são criados.